app/assets/javascripts/oxalis/store.js

```diff
@@ -109,34 +117,21 @@ export type DataLayerType = {
   +category: CategoryType,
   +boundingBox: BoundingBoxObjectType,
   +resolutions: Array<number>,
-  // +fallback: any,
   +elementClass: ElementClassType,
-  +mappings: Array<MappingType>,
+  +mappings?: Array<MappingType>,
 };
 
-export type RestrictionsType = {
-  +allowAccess: boolean,
-  +allowUpdate: boolean,
-  +allowFinish: boolean,
-  +allowDownload: boolean,
+export type SegmentationDataLayerType = DataLayerType & {
+  +largestSegmentId: number,
 };
 
-export type AllowedModeType = "orthogonal" | "oblique" | "flight" | "volume";
+export type RestrictionsType = APIRestrictionsType;
```

Comment:
Do we really need two types for the same thing?

Reply:
Same below.

Reply:
Hmm, not sure whether I introduced this or imitated types that were there before. I think I did it to decouple backend and frontend types, so that when a server type changes, this line can be used to adapt the frontend type to it.
Not sure whether it's worthwhile, we could also get rid of those doubled types.